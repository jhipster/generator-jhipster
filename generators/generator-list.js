/**
 * Copyright 2013-2022 the original author or authors from the JHipster project.
 *
 * This file is part of the JHipster project, see https://www.jhipster.tech/
 * for more information.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

const Generators = {
  GENERATOR_ADD: 'add',
  GENERATOR_APP: 'app',
  GENERATOR_AWS: 'aws',
  GENERATOR_AZURE_APP_SERVICE: 'azure-app-service',
  GENERATOR_AZURE_SPRING_CLOUD: 'azure-spring-cloud',
  GENERATOR_BASE: 'base',
  GENERATOR_BOOTSTRAP: 'bootstrap',
  GENERATOR_BOOTSTRAP_APPLICATION: 'bootstrap-application',
  GENERATOR_CI_CD: 'ci-cd',
  /* @deprecated */
  GENERATOR_CICD: 'ci-cd',
  GENERATOR_CLIENT: 'client',
  GENERATOR_CLOUDFOUNDRY: 'cloudfoundry',
  GENERATOR_COMMON: 'common',
  GENERATOR_CYPRESS: 'cypress',
  GENERATOR_DATABASE_CHANGELOG: 'database-changelog',
  GENERATOR_DATABASE_CHANGELOG_LIQUIBASE: 'database-changelog-liquibase',
  GENERATOR_DOCKER_COMPOSE: 'docker-compose',
  GENERATOR_ENTITIES: 'entities',
  GENERATOR_ENTITIES_CLIENT: 'entities-client',
  GENERATOR_ENTITY: 'entity',
  GENERATOR_ENTITY_CLIENT: 'entity-client',
  GENERATOR_ENTITY_I_18_N: 'entity-i18n',
  /* @deprecated */
  GENERATOR_ENTITY_I18N: 'entity-i18n',
  GENERATOR_ENTITY_SERVER: 'entity-server',
  GENERATOR_EXPORT_JDL: 'export-jdl',
  GENERATOR_GAE: 'gae',
  GENERATOR_GENERATE_BLUEPRINT: 'generate-blueprint',
  GENERATOR_GRADLE: 'gradle',
  GENERATOR_HEROKU: 'heroku',
  GENERATOR_INFO: 'info',
  GENERATOR_INIT: 'init',
  GENERATOR_JAVA: 'java',
  GENERATOR_KUBERNETES: 'kubernetes',
  GENERATOR_KUBERNETES_HELM: 'kubernetes-helm',
  GENERATOR_KUBERNETES_KNATIVE: 'kubernetes-knative',
  GENERATOR_LANGUAGES: 'languages',
  GENERATOR_MAVEN: 'maven',
  GENERATOR_OPENAPI_CLIENT: 'openapi-client',
  GENERATOR_OPENSHIFT: 'openshift',
  GENERATOR_PAGE: 'page',
  GENERATOR_PROJECT_NAME: 'project-name',
  GENERATOR_SERVER: 'server',
  GENERATOR_SPRING_BOOT: 'spring-boot',
  GENERATOR_SPRING_CONTROLLER: 'spring-controller',
  GENERATOR_SPRING_SERVICE: 'spring-service',
  GENERATOR_UPGRADE: 'upgrade',
  GENERATOR_UPGRADE_CONFIG: 'upgrade-config',
  GENERATOR_WORKSPACES: 'workspaces',
};

module.exports = Generators;
