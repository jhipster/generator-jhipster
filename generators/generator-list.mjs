/**
 * Copyright 2013-2023 the original author or authors from the JHipster project.
 *
 * This file is part of the JHipster project; see https://www.jhipster.tech/
 * for more information.
 *
 * Licensed under the Apache License; Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing; software
 * distributed under the License is distributed on an "AS IS" BASIS;
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND; either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

export const GENERATOR_ANGULAR = 'angular';
export const GENERATOR_APP = 'app';
export const GENERATOR_AWS = 'aws';
export const GENERATOR_AZURE_APP_SERVICE = 'azure-app-service';
export const GENERATOR_AZURE_SPRING_CLOUD = 'azure-spring-cloud';
export const GENERATOR_BASE = 'base';
export const GENERATOR_BASE_APPLICATION = 'base-application';
export const GENERATOR_BASE_DOCKER = 'base-docker';
export const GENERATOR_BOOTSTRAP = 'bootstrap';
export const GENERATOR_BOOTSTRAP_APPLICATION = 'bootstrap-application';
export const GENERATOR_BOOTSTRAP_APPLICATION_BASE = 'bootstrap-application-base';
export const GENERATOR_BOOTSTRAP_APPLICATION_CLIENT = 'bootstrap-application-client';
export const GENERATOR_BOOTSTRAP_APPLICATION_SERVER = 'bootstrap-application-server';
export const GENERATOR_CASSANDRA = 'cassandra';
export const GENERATOR_CI_CD = 'ci-cd';
export const GENERATOR_CLIENT = 'client';
export const GENERATOR_CLOUDFOUNDRY = 'cloudfoundry';
export const GENERATOR_COMMON = 'common';
export const GENERATOR_COUCHBASE = 'couchbase';
export const GENERATOR_CYPRESS = 'cypress';
export const GENERATOR_DOCKER = 'docker';
export const GENERATOR_DOCKER_COMPOSE = 'docker-compose';
export const GENERATOR_ELASTICSEARCH = 'elasticsearch';
export const GENERATOR_ENTITIES = 'entities';
export const GENERATOR_ENTITY = 'entity';
export const GENERATOR_EXPORT_JDL = 'export-jdl';
export const GENERATOR_GAE = 'gae';
export const GENERATOR_GENERATE_BLUEPRINT = 'generate-blueprint';
export const GENERATOR_GIT = 'git';
export const GENERATOR_GRADLE = 'gradle';
export const GENERATOR_HEROKU = 'heroku';
export const GENERATOR_INFO = 'info';
export const GENERATOR_INIT = 'init';
export const GENERATOR_KAFKA = 'kafka';
export const GENERATOR_KUBERNETES = 'kubernetes';
export const GENERATOR_KUBERNETES_HELM = 'kubernetes-helm';
export const GENERATOR_KUBERNETES_KNATIVE = 'kubernetes-knative';
export const GENERATOR_LANGUAGES = 'languages';
export const GENERATOR_LIQUIBASE = 'liquibase';
export const GENERATOR_LIQUIBASE_CHANGELOGS = 'liquibase-changelogs';
export const GENERATOR_MAVEN = 'maven';
export const GENERATOR_MONGODB = 'mongodb';
export const GENERATOR_OPENAPI_CLIENT = 'openapi-client';
export const GENERATOR_OPENSHIFT = 'openshift';
export const GENERATOR_PAGE = 'page';
export const GENERATOR_PROJECT_NAME = 'project-name';
export const GENERATOR_REACT = 'react';
export const GENERATOR_SERVER = 'server';
export const GENERATOR_SPRING_CONTROLLER = 'spring-controller';
export const GENERATOR_SPRING_SERVICE = 'spring-service';
export const GENERATOR_UPGRADE = 'upgrade';
export const GENERATOR_VUE = 'vue';
export const GENERATOR_WORKSPACES = 'workspaces';
