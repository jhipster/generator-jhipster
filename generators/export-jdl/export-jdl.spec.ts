import { before, it, describe, expect } from 'esmocha';
import { defaultHelpers as helpers } from '../../test/support/index.js';
import { GENERATOR_EXPORT_JDL } from '../generator-list.js';

const files = {
  '.jhipster/Country.json': {
    fluentMethods: true,
    relationships: [
      {
        relationshipType: 'one-to-one',
        relationshipName: 'region',
        otherEntityName: 'region',
        otherEntityField: 'id',
        relationshipSide: 'left',
        otherEntityRelationshipName: 'country',
      },
    ],
    fields: [
      {
        fieldName: 'countryId',
        fieldType: 'Long',
        documentation: 'The country Id',
      },
      {
        fieldName: 'countryName',
        fieldType: 'String',
      },
    ],
    changelogDate: '20160926101210',
    entityTableName: 'country',
    dto: 'no',
    pagination: 'no',
    service: 'no',
  },
  '.jhipster/Department.json': {
    fluentMethods: false,
    relationships: [
      {
        relationshipType: 'one-to-one',
        relationshipName: 'location',
        otherEntityName: 'location',
        otherEntityField: 'id',
        relationshipSide: 'left',
        otherEntityRelationshipName: 'department',
      },
      {
        relationshipType: 'one-to-many',
        relationshipValidateRules: 'required',
        relationshipName: 'employee',
        relationshipSide: 'left',
        otherEntityName: 'employee',
        otherEntityRelationshipName: 'department',
      },
    ],
    fields: [
      {
        fieldName: 'departmentId',
        fieldType: 'Long',
      },
      {
        fieldName: 'departmentName',
        fieldType: 'String',
        fieldValidateRules: ['required'],
      },
    ],
    changelogDate: '20160926092246',
    entityTableName: 'department',
    dto: 'no',
    pagination: 'no',
    service: 'no',
  },
  '.jhipster/Employee.json': {
    relationships: [
      {
        relationshipSide: 'right',
        relationshipName: 'department',
        otherEntityName: 'department',
        relationshipType: 'many-to-one',
        otherEntityField: 'foo',
        otherEntityRelationshipName: 'employee',
      },
      {
        relationshipType: 'one-to-many',
        relationshipName: 'job',
        otherEntityName: 'job',
        relationshipSide: 'left',
        otherEntityRelationshipName: 'employee',
      },
      {
        relationshipType: 'many-to-one',
        relationshipName: 'manager',
        otherEntityName: 'employee',
        relationshipSide: 'left',
        otherEntityField: 'id',
      },
    ],
    fields: [
      {
        fieldName: 'employeeId',
        fieldType: 'Long',
      },
      {
        fieldName: 'employeeUuid',
        fieldType: 'UUID',
      },
      {
        fieldName: 'firstName',
        documentation: 'The firstname attribute.',
        fieldType: 'String',
      },
      {
        fieldName: 'lastName',
        fieldType: 'String',
      },
      {
        fieldName: 'email',
        fieldType: 'String',
      },
      {
        fieldName: 'phoneNumber',
        fieldType: 'String',
      },
      {
        fieldName: 'hireDate',
        fieldType: 'ZonedDateTime',
      },
      {
        fieldName: 'salary',
        fieldType: 'Long',
        fieldValidateRules: ['min', 'max'],
        fieldValidateRulesMin: 10000,
        fieldValidateRulesMax: 1000000,
      },
      {
        fieldName: 'commissionPct',
        fieldType: 'Long',
      },
    ],
    changelogDate: '20160926083805',
    documentation: 'The Employee entity.',
    entityTableName: 'emp',
    dto: 'mapstruct',
    pagination: 'infinite-scroll',
    service: 'serviceClass',
    fluentMethods: false,
    searchEngine: 'elasticsearch',
    angularJSSuffix: 'myentities',
    microserviceName: 'mymicroservice',
  },
  '.jhipster/Job.json': {
    fluentMethods: true,
    relationships: [
      {
        relationshipName: 'employee',
        otherEntityName: 'employee',
        relationshipType: 'many-to-one',
        relationshipSide: 'right',
        otherEntityField: 'id',
        otherEntityRelationshipName: 'job',
      },
      {
        relationshipType: 'many-to-many',
        otherEntityRelationshipName: 'job',
        relationshipName: 'task',
        otherEntityName: 'task',
        otherEntityField: 'title',
        relationshipSide: 'left',
      },
    ],
    fields: [
      {
        fieldName: 'jobId',
        fieldType: 'Long',
      },
      {
        fieldName: 'jobTitle',
        fieldType: 'String',
      },
      {
        fieldName: 'minSalary',
        fieldType: 'Long',
      },
      {
        fieldName: 'maxSalary',
        fieldType: 'Long',
      },
    ],
    changelogDate: '20160924093047',
    entityTableName: 'job',
    dto: 'no',
    pagination: 'pagination',
    service: 'no',
  },
  '.jhipster/JobHistory.json': {
    fluentMethods: true,
    relationships: [
      {
        relationshipType: 'one-to-one',
        relationshipName: 'job',
        otherEntityName: 'job',
        otherEntityField: 'id',
        relationshipSide: 'left',
        otherEntityRelationshipName: 'jobHistory',
      },
      {
        relationshipType: 'one-to-one',
        relationshipName: 'department',
        otherEntityName: 'department',
        otherEntityField: 'id',
        relationshipSide: 'left',
        otherEntityRelationshipName: 'jobHistory',
      },
      {
        relationshipType: 'one-to-one',
        relationshipName: 'employee',
        otherEntityName: 'employee',
        otherEntityField: 'id',
        relationshipSide: 'left',
        otherEntityRelationshipName: 'jobHistory',
      },
    ],
    fields: [
      {
        fieldName: 'startDate',
        fieldType: 'ZonedDateTime',
      },
      {
        fieldName: 'endDate',
        fieldType: 'ZonedDateTime',
      },
      {
        fieldName: 'language',
        fieldType: 'Language',
        fieldValues: 'FRENCH,ENGLISH,SPANISH',
      },
    ],
    changelogDate: '20160924092444',
    entityTableName: 'job_history',
    dto: 'no',
    pagination: 'infinite-scroll',
    service: 'no',
  },
  '.jhipster/NoEntity.txt': {
    fluentMethods: true,
    relationships: [
      {
        relationshipType: 'one-to-one',
        relationshipName: 'country',
        otherEntityName: 'country',
        otherEntityField: 'id',
        relationshipSide: 'left',
        otherEntityRelationshipName: 'location',
      },
    ],
    fields: [
      {
        fieldName: 'locationId',
        fieldType: 'Long',
      },
      {
        fieldName: 'streetAddress',
        fieldType: 'String',
      },
      {
        fieldName: 'postalCode',
        fieldType: 'String',
      },
      {
        fieldName: 'city',
        fieldType: 'String',
      },
      {
        fieldName: 'stateProvince',
        fieldType: 'String',
      },
    ],
    changelogDate: '20160924092439',
    entityTableName: 'location',
    dto: 'no',
    pagination: 'no',
    service: 'no',
  },
  '.jhipster/Region.json': {
    fluentMethods: true,
    relationships: [
      {
        relationshipType: 'one-to-one',
        relationshipName: 'country',
        otherEntityName: 'country',
        relationshipSide: 'right',
        otherEntityRelationshipName: 'region',
      },
    ],
    fields: [
      {
        fieldName: 'regionId',
        fieldType: 'Long',
      },
      {
        fieldName: 'regionName',
        fieldType: 'String',
      },
    ],
    changelogDate: '20160926083800',
    entityTableName: 'region',
    dto: 'no',
    pagination: 'no',
    service: 'no',
  },
  '.jhipster/Task.json': {
    fluentMethods: true,
    relationships: [
      {
        relationshipType: 'many-to-many',
        relationshipValidateRules: 'required',
        relationshipName: 'job',
        otherEntityName: 'job',
        relationshipSide: 'right',
        otherEntityRelationshipName: 'task',
      },
    ],
    fields: [
      {
        fieldName: 'taskId',
        fieldType: 'Long',
      },
      {
        fieldName: 'title',
        fieldType: 'String',
      },
      {
        fieldName: 'description',
        fieldType: 'String',
      },
    ],
    changelogDate: '20160926100704',
    entityTableName: 'task',
    dto: 'no',
    pagination: 'no',
    service: 'no',
  },
};

const applicationConfig = {
  jhipsterVersion: '3.7.1',
  baseName: 'standard',
  packageName: 'com.mycompany.myapp',
  packageFolder: 'com/mycompany/myapp',
  serverPort: '8080',
  authenticationType: 'session',
  cacheProvider: 'ehcache',
  enableHibernateCache: true,
  websocket: 'no',
  databaseType: 'sql',
  devDatabaseType: 'h2Memory',
  prodDatabaseType: 'postgresql',
  searchEngine: 'no',
  buildTool: 'gradle',
  rememberMeKey: '5f1100e7eae25e2abe32d7b2031ac1f2acc778d8',
  applicationType: 'monolith',
  testFrameworks: [],
  jhiPrefix: 'jhi',
  enableTranslation: true,
  nativeLanguage: 'en',
  languages: ['en'],
  skipClient: true,
  skipServer: true,
};

describe('generator - export-jdl', () => {
  describe('exports entities to a JDL file without argument', () => {
    let runResult;

    before(async () => {
      runResult = await helpers.runJHipster(GENERATOR_EXPORT_JDL).withJHipsterConfig(applicationConfig).withFiles(files).commitFiles();
    });

    it('should match snapshot', () => {
      expect(runResult.getSnapshot(file => file.path.endsWith('.jdl'))).toMatchSnapshot();
    });
    it('creates the jdl file based on app name', () => {
      runResult.assertFile('standard.jdl');
    });
  });

  describe('exports entities to a JDL file with file argument', () => {
    let runResult;

    before(async () => {
      runResult = await helpers
        .runJHipster(GENERATOR_EXPORT_JDL)
        .withJHipsterConfig(applicationConfig)
        .withFiles(files)
        .commitFiles()
        .withArguments('custom-app.jdl');
    });

    it('should match snapshot', () => {
      expect(runResult.getSnapshot(file => file.path.endsWith('.jdl'))).toMatchSnapshot();
    });
    it('creates the jdl file', () => {
      runResult.assertFile('custom-app.jdl');
    });
  });
});
