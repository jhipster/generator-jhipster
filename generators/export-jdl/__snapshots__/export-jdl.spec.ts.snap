// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generator - export-jdl exports entities to a JDL file with file argument should match snapshot 1`] = `
{
  "custom-app.jdl": {
    "contents": "application {
  config {
    applicationType monolith
    authenticationType session
    baseName standard
    buildTool gradle
    cacheProvider ehcache
    creationTimestamp 1577836800000
    databaseType sql
    devDatabaseType h2Memory
    enableHibernateCache true
    enableTranslation true
    jhiPrefix jhi
    jhipsterVersion "3.7.1"
    languages [en]
    nativeLanguage en
    packageName com.mycompany.myapp
    prodDatabaseType postgresql
    rememberMeKey "5f1100e7eae25e2abe32d7b2031ac1f2acc778d8"
    searchEngine no
    serverPort 8080
    skipClient true
    skipServer true
    testFrameworks []
    websocket no
  }

  entities Country, Department, Employee, Job, JobHistory, Region, Task
}

entity Country {
  /**
   * The country Id
   */
  countryId Long
  countryName String
}
entity Department {
  departmentId Long
  departmentName String required
}
/**
 * The Employee entity.
 */
entity Employee (emp) {
  employeeId Long
  employeeUuid UUID
  /**
   * The firstname attribute.
   */
  firstName String
  lastName String
  email String
  phoneNumber String
  hireDate ZonedDateTime
  salary Long min(10000) max(1000000)
  commissionPct Long
}
entity Job {
  jobId Long
  jobTitle String
  minSalary Long
  maxSalary Long
}
entity JobHistory {
  startDate ZonedDateTime
  endDate ZonedDateTime
  language Language
}
entity Region {
  regionId Long
  regionName String
}
entity Task {
  taskId Long
  title String
  description String
}
enum Language {
  FRENCH,
  ENGLISH,
  SPANISH
}

relationship OneToOne {
  Country{region} to Region{country}
  JobHistory{job} to Job
  JobHistory{department} to Department
  JobHistory{employee} to Employee
}
relationship OneToMany {
  Department{employee required} to Employee{department(foo)}
  Employee{job} to Job{employee}
}
relationship ManyToOne {
  Employee{manager} to Employee
}
relationship ManyToMany {
  Job{task(title)} to Task{job required}
}

noFluentMethod Department, Employee
dto Employee with mapstruct
paginate Employee, JobHistory with infinite-scroll
paginate Job with pagination
service Employee with serviceClass
search Employee with elasticsearch
angularSuffix Employee with myentities
microservice Employee with mymicroservice
",
    "stateCleared": "modified",
  },
}
`;

exports[`generator - export-jdl exports entities to a JDL file without argument should match snapshot 1`] = `
{
  "standard.jdl": {
    "contents": "application {
  config {
    applicationType monolith
    authenticationType session
    baseName standard
    buildTool gradle
    cacheProvider ehcache
    creationTimestamp 1577836800000
    databaseType sql
    devDatabaseType h2Memory
    enableHibernateCache true
    enableTranslation true
    jhiPrefix jhi
    jhipsterVersion "3.7.1"
    languages [en]
    nativeLanguage en
    packageName com.mycompany.myapp
    prodDatabaseType postgresql
    rememberMeKey "5f1100e7eae25e2abe32d7b2031ac1f2acc778d8"
    searchEngine no
    serverPort 8080
    skipClient true
    skipServer true
    testFrameworks []
    websocket no
  }

  entities Country, Department, Employee, Job, JobHistory, Region, Task
}

entity Country {
  /**
   * The country Id
   */
  countryId Long
  countryName String
}
entity Department {
  departmentId Long
  departmentName String required
}
/**
 * The Employee entity.
 */
entity Employee (emp) {
  employeeId Long
  employeeUuid UUID
  /**
   * The firstname attribute.
   */
  firstName String
  lastName String
  email String
  phoneNumber String
  hireDate ZonedDateTime
  salary Long min(10000) max(1000000)
  commissionPct Long
}
entity Job {
  jobId Long
  jobTitle String
  minSalary Long
  maxSalary Long
}
entity JobHistory {
  startDate ZonedDateTime
  endDate ZonedDateTime
  language Language
}
entity Region {
  regionId Long
  regionName String
}
entity Task {
  taskId Long
  title String
  description String
}
enum Language {
  FRENCH,
  ENGLISH,
  SPANISH
}

relationship OneToOne {
  Country{region} to Region{country}
  JobHistory{job} to Job
  JobHistory{department} to Department
  JobHistory{employee} to Employee
}
relationship OneToMany {
  Department{employee required} to Employee{department(foo)}
  Employee{job} to Job{employee}
}
relationship ManyToOne {
  Employee{manager} to Employee
}
relationship ManyToMany {
  Job{task(title)} to Task{job required}
}

noFluentMethod Department, Employee
dto Employee with mapstruct
paginate Employee, JobHistory with infinite-scroll
paginate Job with pagination
service Employee with serviceClass
search Employee with elasticsearch
angularSuffix Employee with myentities
microservice Employee with mymicroservice
",
    "stateCleared": "modified",
  },
}
`;
